apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -391,23 +395,41 @@
     </div>
     <% } %>
     <%!
-        private String getRecoverAccountUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            boolean isUsernameRecovery,
-            String urlParameters) {
-
-            return identityMgtEndpointContext + "/recoveraccountrouter.do?" + urlParameters +
-                "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+        private String getRecoverAccountUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                boolean isUsernameRecovery, String urlParameters) {
+
+            return buildUrl(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_RECOVER + "?" + urlParameters
+                    + "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode
+                    .forHtmlAttribute(urlEncodedURL);
+        }
+
+        private String getRegistrationUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                String urlParameters) {
+
+            return buildUrl(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_REGISTER + "?" + urlParameters
+                    + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
         }
 
-        private String getRegistrationUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            String urlParameters) {
+        private String buildUrl(String identityMgtEndpointContext) {
 
-            return identityMgtEndpointContext + "/register.do?" + urlParameters +
-                "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+            String url;
+            if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+                url = buildTenantQualifiedUrl(ACCOUNT_RECOVERY_ENDPOINT);
+            } else if (StringUtils.isNotBlank(identityMgtEndpointContext)) {
+                url = identityMgtEndpointContext;
+            } else {
+                url = getServerURL(ACCOUNT_RECOVERY_ENDPOINT, true, true);
+            }
+            return url;
+        }
+
+        private String buildTenantQualifiedUrl(String context) {
+
+            try {
+                return ServiceURLBuilder.create().addPath(context).build().getAbsoluteURL();
+            } catch (URLBuilderException e) {
+                throw new IdentityRuntimeException("Error while building url for context: " + context);
```

Comment:
This should not be a runtime exception in this context.
Does the JSP handle the error case. Probably the app should have handled exceptions and shown a proper error

Reply:
Also, can this be the absolute URL ? I think the URL returned from here should have been used in front end and observed in browser

Reply:
Updated the PR.